tests/basis_test.py

```diff
@@ -447,10 +763,466 @@ def test_cheb_2nd_call_root_points():
     assert numpy.allclose(f(root_points), numpy.zeros(3))
 
 
+def test_is_complex():
+    """Test that basis functions that output complex values indicate it"""
+    assert smolyay.basis.ChebyshevFirstKind(0)._is_complex == False
+    assert smolyay.basis.ChebyshevSecondKind(0)._is_complex == False
+    assert smolyay.basis.Trigonometric(0)._is_complex == True
+
+
 # Test a set of basis functions
-def test_set_initialize():
-    """Check BasisFunctionSet correctly initializes"""
-    f = smolyay.basis.BasisFunctionSet([])
-    f2 = smolyay.basis.BasisFunctionSet([smolyay.basis.ChebyshevFirstKind(0)])
-    assert f.basis_functions == []
-    assert f2.basis_functions[0].degree == 0
+def test_set_base_class_initialize():
+    """Test BasisFunctionSet correctly initializes"""
+    f = smolyay.basis.BasisFunctionSet([smolyay.basis.ChebyshevFirstKind(0)])
+    assert f.basis_functions[0].degree == 0
+    assert f[0].degree == 0
```

Comment:
This are really just testing if the first element of the set can be accessed. So, instead, save the basis function to a variable and `assert f.basis_functions[0] is bf`